gap/PolygonalComplexes/constructors.gi

```diff
@@ -493,6 +493,44 @@ InstallMethod( UmbrellaDescriptorOfSurface,
         return umbdesc;
 end);
 
+InstallMethod(UmbrellaTipDescriptorOfSurface, 
+    "for a vertexfaithful simplicial surface", 
+    [IsSimplicialSurface],
+    function(surf)
+    local vertex, umbdesc, edge, vertEdges, umbVertices, umbPath;
+    if not IsVertexFaithful(surf) then
+        return fail;
+    fi;
+    umbdesc:=[];
+    for vertex in Vertices(surf) do
+        #UmbrellaTipDescriptor for a vertex
+        umbPath:=UmbrellaPathOfVertex(surf,vertex);
+        vertEdges:=EdgesAsList(umbPath);
+        vertEdges:=List(vertEdges, x -> VerticesOfEdges(surf)[x]);
+        umbVertices:=[];
+        for edge in vertEdges do
+            if not edge[1]=vertex then 
+                Add(umbVertices, edge[1]);
+            else
+                Add(umbVertices, edge[2]);
+            fi;
+        od;
+
+        #Adding the cycles/lists
+        if IsClosedPath(umbPath) then
+            #umbpath is a closed umbrella
+            Remove(umbVertices);
+            Add(umbdesc, CycleFromList(umbVertices));
+        else
+            #umbpath is not a closed umbrella
+            Add(umbVertices, umbVertices[1]);
```

Comment:
Why do you move the first entry to the end? The two boundary vertices must be at the end and in the beginning, which isn't the case here (as I can see in the example).

Reply:
Oh, that is true. For some reason I thought the lists should be ordered and then the first vertex should be at the end, I guess. I'll change it.